codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -277,6 +280,13 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
             }
 
             if (stream == null || stream.isResetSent() || streamCreatedAfterGoAwaySent(streamId)) {
+                if (logger.isInfoEnabled()) {
```

Comment:
I need to review this code a bit and see if we can do better ... the situation I was in where header frames were out of order was falling into here and was silently dropped (and subsequent data frames I believe). A `protocol error` would have made it much easier to track down, but I know there were reasons why we couldn't tell with certainty if it was an error or not.

https://tools.ietf.org/html/rfc7540#section-5.1.1
> The identifier of a newly established stream MUST be numerically
greater than all streams that the initiating endpoint has opened or
reserved. This governs streams that are opened using a HEADERS frame
and streams that are reserved using PUSH_PROMISE. An endpoint that
receives an unexpected stream identifier MUST respond with a
connection error (Section 5.4.1) of type PROTOCOL_ERROR.